x-pack/plugins/security_solution/public/management/pages/policy/view/policy_advanced.tsx

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+import { EuiFieldText, EuiFlexItem, EuiFlexGroup } from '@elastic/eui';
+import { cloneDeep } from 'lodash';
+import { policyConfig } from '../store/policy_details/selectors';
+import { usePolicyDetailsSelector } from './policy_hooks';
+
+function setValue(obj: Record<string, unknown>, value: string, path: string[]) {
+  let newPolicyConfig = obj;
+  for (let i = 0; i < path.length - 1; i++) {
+    if (!newPolicyConfig[path[i]]) {
+      newPolicyConfig[path[i]] = {} as Record<string, unknown>;
+    }
+    newPolicyConfig = newPolicyConfig[path[i]] as Record<string, unknown>;
+  }
+  newPolicyConfig[path[path.length - 1]] = value;
+}
+
+function getValue(obj: Record<string, unknown>, path: string[]) {
+  let currentPolicyConfig = obj;
+
+  for (let i = 0; i < path.length - 1; i++) {
+    if (currentPolicyConfig[path[i]]) {
+      currentPolicyConfig = currentPolicyConfig[path[i]] as Record<string, unknown>;
+    } else {
+      return undefined;
+    }
+  }
+  return currentPolicyConfig[path[path.length - 1]];
+}
+
+export const PolicyAdvanced = React.memo(
+  ({
+    configPath,
+    firstSupportedVersion,
+    lastSupportedVersion,
+  }: {
+    configPath: string[];
+    firstSupportedVersion: string;
+    lastSupportedVersion?: string;
+  }) => {
+    const dispatch = useDispatch();
+    const policyDetailsConfig = usePolicyDetailsSelector(policyConfig);
+    const onChange = useCallback(
+      (event) => {
+        if (policyDetailsConfig) {
+          const newPayload = cloneDeep(policyDetailsConfig);
+          setValue(
+            (newPayload as unknown) as Record<string, unknown>,
+            event.target.value,
+            configPath
+          );
+          dispatch({
+            type: 'userChangedPolicyConfig',
+            payload: { policyConfig: newPayload },
+          });
+        }
+      },
+      [dispatch, policyDetailsConfig, configPath]
+    );
+
+    const value =
+      policyDetailsConfig &&
+      getValue((policyDetailsConfig as unknown) as Record<string, unknown>, configPath);
+
+    return (
+      <>
+        <EuiFlexGroup>
```

Comment:
I tried this so that each row would be in its own group, since from my understanding of the docs a group would span one row of the page. But more than one group is ending up in a row:
![image](https://user-images.githubusercontent.com/57721870/96631669-9d8c6780-12e4-11eb-9385-f81d3f1e9dc9.png)

If I take out the `EuiFlexGroup`s and just use `EuiFlexGrid columns={3}` to wrap the entire thing, it looks better:
![image](https://user-images.githubusercontent.com/57721870/96631751-bc8af980-12e4-11eb-84da-2939ff1f527e.png)

but doesn't handle a change to the page size gracefully:
![image](https://user-images.githubusercontent.com/57721870/96631782-c7458e80-12e4-11eb-9ceb-5ffa5f17139e.png)

@parkiino @paul-tavares @kevinlog any thoughts on how to get each schema item in its own row so that the row will also resize gracefully?

Reply:
@jmiller263 

I chatted with @bfishel and @caitlinbetz and we could go with another design here which will allow us to skip using the flex grid all together.  It will use the `EuiFormRow` component and use form labels to take care of the other pieces.  The responsiveness will also be much nicer and we won't have to deal with it directly ourselves.  In the `advanced_policy.tsx` file, you could replace the EuiFlex components with the below.

```
        <EuiFormRow
          fullWidth
          label={configPath.join('.')}
          labelAppend={
            <EuiText size="xs">
                {lastSupportedVersion
                  ? `${firstSupportedVersion}-${lastSupportedVersion}`
                  : `${firstSupportedVersion}+`}
            </EuiText>
        }>
          <EuiFieldText fullWidth value={value as string} onChange={onChange} />
      </EuiFormRow>
```

And the make sure you remove the `EuiFlexGrid` and Flex components from the parent component.

You end up with something that looks like this:
![image](https://user-images.githubusercontent.com/56395104/96644348-f36a0b00-12f6-11eb-82e0-0fbee910be58.png)

Reply:
@kevinlog: added in 7c79ecec15b75b825c5f60d6e1ca8b13fb700007, thanks for the help!